### PR TITLE
C++ StackWatermark

### PR DESCRIPTION
**Pull request by Lucien950**

### CODE REVIEW

File: firmware/quintuna/CRIT/src/cubemx/Src/main.c [hunk 1]
```diff
@@ -24,6 +24,7 @@
 /* USER CODE BEGIN Includes */
 #include "tasks.h"
 #include "hw_error.h"
+#include "../../../../../shared/src/io/io_log.h"
```
Comment (Lucien950) [inline firmware/quintuna/CRIT/src/cubemx/Src/main.c:27]:
```suggestion
```

File: firmware/quintuna/CRIT/src/app/app_stackMonitorConfig.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include "app_stackMonitor.h"
+
+#include "io_stackMonitorConfig.h"
+
+namespace app
+{
+inline StackMonitor<5> crit_stack_monitor{
```
Comment (liam-ilan) [inline firmware/quintuna/CRIT/src/app/app_stackMonitorConfig.h:8]: Nit - I think it's cleaner to specify the `stack_monitor_config` type explicitly for all items.

**Review by liam-ilan — CHANGES_REQUESTED** [review]
Looks good to me!

**Review by AmirTajaddodi — APPROVED** [review]
LGTM!!

**Pull request merged**